### ISSUE TITLE
Necessary permissions for the Microsoft Graph provider are unclear

### ISSUE THREAD

**Issue by tarkatronic** [microsoft/opensource-portal#161]
**Describe the bug**
I've managed to get this so that I can log in via AAD. Now, when I attempt to link my GitHub account, the portal is returning a 500 error on the `/link` url due to `Invalid status code: 403` from the backend. I've added a bit of extra debug in and I can see this is coming from the Microsoft Graph API:

```js
{
  error: {
    code: 'Authorization_RequestDenied',
    message: 'Insufficient privileges to complete the operation.',
    innerError: {
      date: '2020-08-03T20:30:06',
      'request-id': '994aeeab-f00a-484b-b591-0d8ef14d21cf'
    }
  }
}
```

The problem is, I can't suss out what the necessary permissions are. So far, I have granted delegated permissions for:
* email
* Mail.Read
* openid
* profile
* User.Read

From what I can tell, this should be *more* than enough, given that the request is asking for:
```
?$select=id,mailNickname,userType,displayName,givenName,mail,userPrincipalName
```

What am I missing from my permissions to get this working properly? Will this even work for a non-MS company?

**Comment by tarkatronic** [comment]
I believe I have managed to find the answer here, thanks to the [Microsoft Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer). When executing a query against `/v1.0/users/<id>/`, it shows 15 permissions in use. In actuality, once I de-duped those permissions, it narrowed down to 8.

* User.Read
* User.Read.All
* User.ReadBasic.All
* User.ReadWrite
* User.ReadWrite.All
* Directory.AccessAsUser.All
* Directory.Read.All
* Directory.ReadWrite.All

This does not changed based on what fields you are querying for.
So my question now is... do I really need to add all of these permissions?? This seems quite excessive just to grab some information about my own profile. Three of these are for *global write* access, and 5 of them require admin consent! Is there no simpler way to retrieve this? This level of permission makes this start to look like a security risk.

As a notes, accessing `/v1.0/me/` requires the same de-duped set of permissions.

**Comment by jeffwilcox** [comment]
There's def. no reason for any kind of write permission!

**Comment by tarkatronic** [comment]
I agree! Unfortunately I'm still getting the same error with my permissions set up like this:
![Screen Shot 2020-08-04 at 5 26 51 PM](https://user-images.githubusercontent.com/37809/89346984-ef273d80-d677-11ea-95fa-61e93ebf2987.png)

I can ask our admins to grant those other permissions, but I really would rather not have to...

**Comment by jeffwilcox** [comment]
I hear you... permissions can take ages to get approved here, too, if at all...

Do you think you need the graph features? Things it would give you:

- ability to block Guest users in the tenant (probably important)
- lookup managers, to notify them if you like that, for new repos etc.

Otherwise, we could probably feature flag it out, or find other options... for example, you may have enough info in the AAD passport response to determine if you think they'll work in your directory.

**Comment by tarkatronic** [comment]
I think this may require a combination of solutions. I can't say for sure whether I will need the graph features, but those two uses you listed sure do sound nice to have.

Regardless, I think a feature flag would be good here. That way, the application will at least work by default. And aside from that, it's probably good to add some documentation on these permission requirements, to make this setup easier for any other users.

Happy to help out here however I can. 😄

**Comment by ElenaForester** [comment]
I had a similar issue and it was resolved when added `Directory.Read.All` with `Application` type
![image](https://user-images.githubusercontent.com/20944292/126440852-64e52867-b5c6-4318-951c-365d4531bfe8.png)

**Comment by RFOF-NETWORK** [comment]
We are only dealing with permissions from an IT perspective; a semantic catalysis of the respective variations of meanings must be carried out to ensure user-friendliness Let me reveal our method of guaranteeing this: quite simply, using top, middle, and bottom, in which different positions/permissions then compact themselves within each other/mutually for developers.. but to simplify massively, or to define the meaning of "entitled" so clearly that misinterpretation is impossible, be it for back, front, or full stack. If you wish, I can send you here, or wherever you deem appropriate, our innovative open-source graphic based on a vector principle that applies to all types of permissions, like a kind of DNA They only work if they are designed for permissions.
